registry/storage/driver/s3-aws/s3.go

```diff
@@ -787,50 +789,52 @@ func (d *driver) statList(ctx context.Context, s3Path string) (*storagedriver.Fi
 	if err != nil {
 		return nil, err
 	}
-	var fi storagedriver.FileInfoFields
 	if len(resp.Contents) == 1 {
 		if *resp.Contents[0].Key != s3Path {
-			fi.IsDir = true
-			return &fi, nil
+			return &storagedriver.FileInfoFields{
+				Path:  path,
+				IsDir: true,
+			}, nil
 		}
-		fi.IsDir = false
-		fi.Size = *resp.Contents[0].Size
-		fi.ModTime = *resp.Contents[0].LastModified
-		return &fi, nil
+		return &storagedriver.FileInfoFields{
+			Path:    path,
+			Size:    *resp.Contents[0].Size,
+			ModTime: *resp.Contents[0].LastModified,
+		}, nil
 	}
 	if len(resp.CommonPrefixes) == 1 {
-		fi.IsDir = true
-		return &fi, nil
+		return &storagedriver.FileInfoFields{
+			Path:  path,
+			IsDir: true,
+		}, nil
```

Comment:
I can't find a reason in my head to justify all of these add extra lines of code. But maybe you'll persuade me.

Reply:
Formatting aside (these shorter ones can definitely be a 1-line), it's clear at one glance what's returned; previously setting these fields was sprinkled across multiple locations, and even spread across multiple functions, making it more error-prone, and more difficult to see what values are used.